components/form-builder/app/responses/Dialogs/DownloadDialog.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useEffect } from "react";
+import { Dialog, useDialogRef } from "../../shared";
+import { useTranslation } from "react-i18next";
+import { Button } from "@components/globals";
+import { logMessage } from "@lib/logger";
+import axios from "axios";
+import { DownloadFormat } from "@lib/responseDownloadFormats/types";
+import JSZip from "jszip";
+
+export const DownloadDialog = ({
+  checkedItems,
+  isVisible,
+  setIsVisible,
+  downloadError,
+  setDownloadError,
+  formId,
+  onSuccessfulDownload,
+}: {
+  checkedItems: Map<string, boolean>;
+  isVisible: boolean;
+  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  downloadError: boolean;
+  setDownloadError: React.Dispatch<React.SetStateAction<boolean>>;
+  formId: string;
+  onSuccessfulDownload: () => void;
+}) => {
+  const dialogRef = useDialogRef();
+  const { t } = useTranslation("form-builder-responses");
+  const [selectedFormat, setSelectedFormat] = React.useState<DownloadFormat>();
+  const [zip, setZip] = React.useState<boolean>(true);
+
+  useEffect(() => {
+    if (selectedFormat === DownloadFormat.HTML_ZIPPED) {
+      setZip(true);
+    }
+  }, [selectedFormat]);
+
+  if (downloadError) {
+    setDownloadError(false);
```

Comment:
This seems odd -- if there's a download error set the download error to false?

Reply:
It's resetting the download error when the dialog pops. Would this be better in a useEffect or something? It's basically if for some reason you attempted but got a download error, if you pop the dialog again it clears the error.

Reply:
Fixed underlying bug